platform/darwin/scripts/generate-style-code.js

```diff
@@ -187,14 +187,22 @@ global.propertyType = function (property, _private) {
     return _private ? `id <MGLStyleAttributeValue, MGLStyleAttributeValue_Private>` : `id <MGLStyleAttributeValue>`;
 };
 
+global.initLayerIdentifierOnly = function (layerType) {
+    return `_layer = new mbgl::style::${camelize(layerType)}Layer(layerIdentifier.UTF8String);`
+}
+
 global.initLayer = function (layerType) {
     if (layerType == "background") {
-        return `_layer = new mbgl::style::${camelize(layerType)}Layer(layerIdentifier.UTF8String);`
+       return `_layer = new mbgl::style::${camelize(layerType)}Layer(layerIdentifier.UTF8String);` 
```

Comment:
I think we need to take a step back and address this leak in #6254. Until then, any patches will only add to the confusion.